src/components/MoneyRequestHeader.tsx

```diff
@@ -231,7 +240,32 @@ function MoneyRequestHeader({
                     policy={policy}
                     shouldShowBackButton={shouldUseNarrowLayout}
                     onBackButtonPress={onBackButtonPress}
-                />
+                >
+                    {shouldShowMarkAsCashButton && !shouldUseNarrowLayout && (
+                        <Button
+                            success
+                            medium
+                            text={translate('iou.markAsCash')}
+                            style={[styles.p0]}
+                            onPress={() => {
+                                TransactionActions.markAsCash(transaction?.transactionID ?? '', transaction?.reportID ?? '');
```

Comment:
@BrtqKr I think this is the source of your issue. We should be using the transactionThreadID here, not the report ID of the transaction.

One can see this here:
<img width="1650" alt="Screenshot 2024-05-21 at 11 48 06 AM" src="https://github.com/Expensify/App/assets/4741899/39e6f2f6-3b0f-46cc-ba0c-3f8ae3574bc4">
<img width="659" alt="Screenshot 2024-05-21 at 11 48 13 AM" src="https://github.com/Expensify/App/assets/4741899/6eb89c06-6515-4221-94c8-0c8ca6543741">